crypto/bn/bn_print.c

```diff
@@ -23,12 +23,9 @@ char *BN_bn2hex(const BIGNUM *a)
     char *buf;
     char *p;
 
-    if (a->neg && BN_is_zero(a)) {
-        /* "-0" == 3 bytes including NULL terminator */
-        buf = OPENSSL_malloc(3);
-    } else {
-        buf = OPENSSL_malloc(a->top * BN_BYTES * 2 + 2);
-    }
+    if (BN_is_zero(a))
+        return OPENSSL_strdup("0");
```

Comment:
Nit pick: would it make sense to return `"00"`?  Considering we otherwise always give back hex numbers as pairs of nibbles, it seems odd that we'd make an exception for the zero.